### PR TITLE
teip: init at 1.2.1

### PR DESCRIPTION
**Pull request by joefiorini**
###### Motivation for this change

This module is a very useful, general purpose Unix tool that I think others would like to install via nix.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -3132,6 +3132,12 @@
     githubId = 7385287;
     name = "Lana Black";
   };
+  greymd = {
```
Comment (sikmir) [inline maintainers/maintainer-list.nix:3135]: IIRC, adding yourself as a maintainer is preferred to be a separate commit.

Something like, maintainers: add greymd. Then apply the teip commit.

File: pkgs/tools/misc/teip/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{ lib, fetchFromGitHub, rustPlatform, installShellFiles }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "teip";
+  version = "1.2.1";
+
+  src = fetchFromGitHub {
+    owner = "greymd";
+    repo = "teip";
+    rev = "52b7a8b4a6cbf354a3620936cec7add257e2f483";
```
Comment (sikmir) [inline pkgs/tools/misc/teip/default.nix:10]:
```suggestion
    rev = "v${version}";
```

File: pkgs/tools/misc/teip/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  nativeBuildInputs = [ installShellFiles ];
+
+	# Disabling tests because criterion build was hanging for me
+	# on Darwin
+  doCheck = false;
+
+  features = [ "oniguruma" ];
+
+  preFixup = ''
```
Comment (sikmir) [inline pkgs/tools/misc/teip/default.nix:26]:
```suggestion
  postInstall = ''
```

File: pkgs/tools/misc/teip/default.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+  meta = with lib; {
+    description = "Highly efficient \"Masking tape\" for standard input";
+    longDescription = ''
+      Select partial standard input and replace with the result of another command efficiently
+    '';
+    homepage = "https://github.com/greymd/teip";
+    license = with licenses; [ mit ];
+    maintainers = with maintainers; [ greymd ];
+    platforms = platforms.x86_64;
```
Comment (doronbehar) [inline pkgs/tools/misc/teip/default.nix:40]: Are you sure this won't build on other platforms? Also, I think this should be `x86_64-linux` or perhaps in your case I'd set `platforms.linux`.

File: pkgs/tools/misc/teip/default.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    installManPage "$src/man/teip.1"
+
+    installShellCompletion --zsh $src/completion/zsh/_teip
+  '';
+
+  meta = with lib; {
+    description = "Highly efficient \"Masking tape\" for standard input";
+    longDescription = ''
+      Select partial standard input and replace with the result of another command efficiently
+    '';
```
Comment (doronbehar) [inline pkgs/tools/misc/teip/default.nix:36]: I'd put in `description` what you put in `longDescription`, "Masking tape" might be a confusing description for some people.

File: pkgs/tools/misc/teip/default.nix [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  };
+
+  cargoSha256 = "165f9dxggyfagl3pb84hbi898yrwhdbmx4z52fh393a3s965nwpn";
+
+  cargoPatches = [ ./Cargo.lock.patch ];
+
+  nativeBuildInputs = [ installShellFiles ];
+
+	# Disabling tests because criterion build was hanging for me
+	# on Darwin
```
Comment (doronbehar) [inline pkgs/tools/misc/teip/default.nix:21]:
```suggestion
  # Disabling tests because criterion build was hanging for me
  # on Darwin
```

**Review by doronbehar — CHANGES_REQUESTED** [review]

**Comment by doronbehar** [comment]
@joefiorini please address the review comments.

File: pkgs/tools/misc/teip/default.nix [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  doCheck = false;
+
+  features = [ "oniguruma" ];
+
+  preFixup = ''
+    installManPage "$src/man/teip.1"
+
+    installShellCompletion --zsh $src/completion/zsh/_teip
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/teip/default.nix:29]: bash, fish?
